packages/typespec-vscode/test/extension/common/commonSteps.ts

```diff
@@ -0,0 +1,343 @@
+ * folderName: The text in the top input box is usually the current open root directory,
+ * command: After the top input box pops up, the command to be executed
+ */
+async function startWithCommandPalette(
+  page: Page,
+  { folderName, command }: { folderName: string; command: string }
+) {
+  await sleep(2)
+  await page.locator("li").filter({ hasText: folderName }).first().click()
+  await sleep(2)
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/open_top_panel.png") })
+  await page
+    .getByRole("textbox", { name: "Search files by name (append" })
+    .first()
+    .fill(`>TypeSpec: ${command}`)
+  let listForCreate: Locator
+  await retry(
+    page,
+    5,
+    async () => {
+      listForCreate = page
+        .locator("a")
+        .filter({ hasText: `TypeSpec: ${command}` })
+        .first()
+      return (await listForCreate.count()) > 0
+    },
+    "Failed to find the specified option"
+  )
+  await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/input_command.png") })
+  await listForCreate!.click()
+}
+
+/**
+ * Start the Project with Right click on the file
+ * @param page vscode object
+ * @param command create, emit or import
+ * @param type specify whether the click is on file, folder or empty folder
+ * command: specify which command to execute to the project
+ */
+async function startWithRightClick(page: Page, command: string, type?: string) {
+  if (
+    command == "Emit from TypeSpec" ||
+    command == "Preview API Documentation"
+  ) {
+    const target = page.getByRole("treeitem", { name: "main.tsp" }).locator("a")
+    await target.click({ button: "right" })
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/click_main.png") })
+    await page.getByRole("menuitem", { name: command }).click()
+    await page.screenshot({ path: path.resolve(__dirname, `../../images-linux/` +
+      `${command == "Emit from TypeSpec" ? "emit" : "preview"}_typespec.png`)
+    })
+  } else if (command == "Import TypeSpec from Openapi 3") {
+    const targetName =
+      type === "emptyfolder"
+        ? "ImportTypespecProjectEmptyFolder"
+        : "openapi.3.0.yaml"
+    const target = page.getByRole("treeitem", { name: targetName }).locator("a")
+    await target.click({ button: "right" })
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/openapi.3.0.png") })
+    await sleep(3)
+    await page
+      .getByRole("menuitem", { name: "Import TypeSpec from OpenAPI" })
+      .click()
+    await page.screenshot({ path: path.resolve(__dirname, "../../images-linux/import_typespec.png") })
+  }
+}
+
+/**
+ * In vscode, when you need to select a folder or a file, call this method
+ * @param file When selecting a file, just pass it in. If you need to select a folder, you do not need to pass this parameter in.
+ */
+async function selectFolder(page: Page, file: string = "") {
+  await sleep(10)
+  await keyboard.type(file)
```

Comment:
can you not use `page.keyboard` instead of this additional package https://playwright.dev/docs/api/class-keyboard

Reply:
Since `page.keyboard` cannot simulate keyboard operations outside of vscode, for example, selecting a target folder, this scene will jump out of vscode.

Reply:
hhm this dependency doesn't have much usage and is loading a lot of other dependency. This is quite a lot just for this tiny thing.

Can you check with my opther comment that if you call directly the vscode executeCommand api you cannot pass the folder as input and skip those steps.

If not can making those basic change not be simply emulate or using a smaller library with more usage.